src/data_generators/m3_data_design.py

```diff
@@ -157,20 +166,198 @@
             "order": OrderOptions.random,
             "correlated_sets": CorrelatedSetsOptions.one,
             "shared_prop": 0.75,
-            "random_generator": 8,
-        },
-    ),
-    GeneratorParameters(
-        "Copula",
-        CopulaDataSet,
-        {
-            "largest_pub_to_universe_ratio": 0.5,
-            "copula_generator": GaussianCopula(corr=0),
-            "random_generator": 1,
+            "random_generator": np.random.default_rng(RNG.integers(1e9)),
         },
     ),
 ]
 
+
+class CopulaCorrelationMatrixGenerator:
+    @staticmethod
+    def homogeneous(p: int, rho: float) -> np.ndarray:
+        """Generate a homogeneous correlation matrix.
+
+        Args:
+            p:  Number of pubs.
+            rho:  Homogenous correlation.
+
+        Returns:
+            A p * p matrix with diagonals being 1 and all off-diagonals being rho.
+        """
+        return splinalg.toeplitz([1] + [rho] * (p - 1))
+
+    @staticmethod
+    def autoregressive(p: int, rho: float) -> np.ndarray:
+        """Generate a autoregressive correlation matrix.
+
+        Args:
+            p:  Number of pubs.
+            rho:  Homogenous correlation.
+
+        Returns:
+            A <p * p> matrix C where C[i, j] = rho^|i - j| for any i, j.
+        """
+        return splinalg.toeplitz(np.power(np.array([rho] * p), np.arange(p)))
+
+    @staticmethod
+    def random(
+        p: int, rng: np.random.Generator = np.random.default_rng(0)
+    ) -> np.ndarray:
+        """Randomly, uniformly draw a correlation matrix.
+
+        Given the dimension, all the possible correlation matrices form a space,
+        and there is a geometric measure on this space.  The geoemetric measure
+        defines a probability space.  This function uniformly draws a correlation
+        matrix from this natural probability space.
+
+        We implemented the uniform sampling algorithm in Section 3.2 of:
```

Comment:
Delete "Section 3.2 in" because it is used to locate the algorithm in the other reference published in 2009.